xlators/storage/posix/src/posix-entry-ops.c

```diff
@@ -1016,6 +1061,13 @@ posix_mkdir(call_frame_t *frame, xlator_t *this, loc_t *loc, mode_t mode,
     posix_set_parent_ctime(frame, this, par_path, -1, loc->parent, &postparent);
 
     op_ret = 0;
+    if (atomic_creation) {
+        op_ret = sys_rename(real_path, orig_real_path);
+        if (op_ret)
+            gf_log(this->name, GF_LOG_ERROR,
+                   "rename is failing old_path %s new_path %s", real_path,
+                   orig_real_path);
+    }
```

Comment:
This seems common code. It should be moved to a function.

Reply:
done